assets/multi-tenant/manifest.yaml

```diff
@@ -0,0 +1,81 @@
+# yaml-language-server: $schema=manifest.json
+---
+
+# model
+model:
+  version: 3
+
+# object type definitions
+types:
+  # user represents a user that can be granted role(s)
+  user:
+    relations:
+      manager: user
+
+    permissions:
+      in_management_chain: manager | manager->in_management_chain
+
+  # group represents a collection of users and/or (nested) groups
+  group:
+    relations:
+      member: user | group#member
+
+  # identity represents a collection of identities for users
+  identity:
+    relations:
+      identifier: user
+
+  # system represents the entire application and grants access to all tenants
+  system:
+    relations:
+      admin: user | group#member
+      editor: user | group#member
+      viewer: user | group#member
+
+    permissions:
+      can_create_tenant: admin
+
+  # tenant represents a tenant in a multi-tenant system
+  tenant:
+    relations:
+      # system that the tenant is part of.
+      system: system
+      owner: user
+      admin: user | group#member
+      editor: user | group#member
+      viewer: user | group#member
+
+    permissions:
+      # group members into 3 high-level categories: can_administer, can_edit, can_view.
+      # these are used to define fine-grained permission.
+      can_administer: owner | admin | system->admin
+      can_edit: editor | can_administer | system->editor
+      can_view: viewer | can_edit | system->viewer
+
+      # fine-grained permissions make it easier to change access levels without
+      # modifying application logic.
+      can_delete_tenant: owner | system->admin
+      can_manage_roles: can_administer
+      can_list_members: can_view
+      # an owner cannot leave the tenant. they must be removed by another owner.
+      can_leave_tenant: can_view - owner
```

Comment:
doesn't this imply another permission - `can_remove_member`?

Reply:
How about renaming `can_manage_roles` to `can_manage_members`? I meant for that permissions to cover adding/removing/updating tenant members.